Tests/Marketplace/search_and_install_packs.py

```diff
@@ -124,102 +108,51 @@ def create_dependencies_data_structure(response_data: dict, dependants_ids: list
         create_dependencies_data_structure(response_data, next_call_dependants_ids, dependencies_data, checked_packs)
 
 
-def get_pack_dependencies(client: demisto_client, pack_data: dict, lock: Lock):
-    """ Get the pack's required dependencies.
+def get_pack_dependencies(client: demisto_client, pack_id: str, lock: Lock) -> dict | None:
+    """
+    Get the pack's required dependencies.
 
     Args:
         client (demisto_client): The configured client to use.
-        pack_data (dict): Contains the pack ID and version.
+        pack_id (str): ID of the pack to get dependencies for.
         lock (Lock): A lock object.
+
     Returns:
-        (list) The pack's dependencies.
+        dict: API response data for the /search/dependencies endpoint.
     """
     global SUCCESS_FLAG
-    pack_id = pack_data['id']
-    logging.debug(f'Getting dependencies for pack {pack_id}')
+
+    api_endpoint = "/contentpacks/marketplace/search/dependencies"
+    body = [{"id": pack_id}]  # 'version' key can be specified alongside "id". Not including it will fetch the latest version.
+
+    logging.debug(f"Fetching dependencies for pack '{pack_id}'.\n"
+                  f"Sending POST request to {api_endpoint} with body: {json.dumps(body)}")
+
     try:
-        response_data, status_code, _ = demisto_client.generic_request_func(
+        response_data, _, _ = demisto_client.generic_request_func(
             client,
-            path='/contentpacks/marketplace/search/dependencies',
+            path=api_endpoint,
             method='POST',
-            body=[pack_data],
+            body=body,
             accept='application/json',
             _request_timeout=None,
-            response_type='object'
+            response_type='object',
         )
-        if 200 <= status_code < 300:
-            dependencies_data: list = []
-            dependants_ids = [pack_id]
-            response_data = response_data.get('dependencies', [])
-            create_dependencies_data_structure(response_data, dependants_ids, dependencies_data, dependants_ids)
-            if dependencies_data:
-                dependencies_str = ', '.join([dep['id'] for dep in dependencies_data])
-                logging.debug(f'Found the following dependencies for pack {pack_id}: {dependencies_str}')
-            return dependencies_data
-        if status_code == 400:
-            logging.error(f'Unable to find dependencies for {pack_id}.')
-            return []
-        msg = response_data.get('message', '')
-        raise Exception(f'status code {status_code}\n{msg}\n')
```

Comment:
Removed this section since `demisto_client.demisto_api.rest.py` raises an APIException if an error code is received on `RESTClientObject.request()`:
```python
        if not 200 <= r.status <= 299:
            raise ApiException(http_resp=r)
```

So in case an error code is returned, it skips straight to `except ApiException as api_ex:` making this code section redundant.

Reply:
`if not 200 <= r.status <= 299:` so need to be checked `if 200 <= status_code < 300`

Reply:
You're right, shouldn't have removed this section:
```python
        if 200 <= status_code < 300:
            dependencies_data: list = []
            dependants_ids = [pack_id]
            response_data = response_data.get('dependencies', [])
            create_dependencies_data_structure(response_data, dependants_ids, dependencies_data, dependants_ids)
            if dependencies_data:
                dependencies_str = ', '.join([dep['id'] for dep in dependencies_data])
                logging.debug(f'Found the following dependencies for pack {pack_id}: {dependencies_str}')
            return dependencies_data
```
I'll revert that